scout/adapter/client.py

```diff
@@ -54,11 +55,12 @@ def get_connection(
             log_uri = uri = "mongodb://%s:%s" % (host, port)
 
     LOG.info("Try to connect to %s" % log_uri)
+
+    client = MongoClient(uri, serverSelectionTimeoutMS=timeout)
     try:
-        client = MongoClient(uri, serverSelectionTimeoutMS=timeout)
-    except ServerSelectionTimeoutError as err:
-        LOG.warning("Connection Refused")
-        raise ConnectionFailure
+        client.server_info()
```

Comment:
This is the actual step that checks if the connection is valid. If it crashes it's not a valid connection (check [here](https://stackoverflow.com/questions/30539183/how-do-you-check-if-the-client-for-a-mongodb-instance-is-valid))

Reply:
NIce!